src/core/comm/coll.cc

```diff
@@ -245,19 +245,21 @@ int collInit(int argc, char* argv[])
   int provided, init_flag = 0;
   CHECK_MPI(MPI_Initialized(&init_flag));
   if (!init_flag) {
-    log_coll.fatal(
+    // TODO: check if -ll:networks none is enabled
+    log_coll.warning(
       "MPI has not been initialized, it should be initialized by "
-      "the networking backend");
+      "the networking backend, if -ll:networks none is passed, "
+      "then this warning can be safely ignored.");
+    int provided;
+    MPI_Init_thread(0, 0, MPI_THREAD_MULTIPLE, &provided);
```

Comment:
I think you need to finalize MPI in collFinalize if it is initialized by collInit.

Reply:
good catch!, thank you